### DISCUSSION TITLE
After changing to pytorch-lightning 1.5.2, omitting the argument of trainer.test() does not work.

### DISCUSSION THREAD

**Discussion by Keiku** [Lightning-AI/pytorch-lightning#10747] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
In case of pytorch-lightning 1.5.2, omission of argument of `trainer.test()` does not work.

```python
        trainer.fit(model, datamodule)
        trainer.test()
```

If it is below, it worked fine. Is this the expected behavior?

```python
        trainer.fit(model, datamodule)
        trainer.test(model, datamodule.test_dataloader())
```

I'm currently refactoring the following:

Keiku/PyTorch-Lightning-CIFAR10: "Not too complicated" training code for CIFAR-10 by PyTorch Lightning https://github.com/Keiku/PyTorch-Lightning-CIFAR10

**Comment by rohitgr7** [comment, marked as answer]
yes, you need to pass the datamodule. I think this was the case after v1.5 and not just v1.5.2.

**Reply by Keiku** [reply]
Thank you for your answering. I will continue to catch up on pytorch lighting.

**Reply by rohitgr7** [reply]
also, you can just pass in 
```py
trainer.test(model, datamodule=datamodule)
```
not need to call `.test_dataloader()` explicitly.
